internal/transport/transport_test.go

```diff
@@ -2592,3 +2592,20 @@ func TestConnectionError_Unwrap(t *testing.T) {
 		t.Error("ConnectionError does not unwrap")
 	}
 }
+
+func TestServerOperateHeaderOnEvenStreamID(t *testing.T) {
```

Comment:
This is a nice approach!

Reply:
But maybe the test name should mention fails to operate somewhere? Also pls add a comment on the test for readability?